File: Agile Development Practices/Code/DotNet/videoworld/UnitTests/Models/ChildrensPriceTest.cs
using NUnit.Framework;
using VideoWorld.Models;

namespace UnitTests.Models
{
    class ChildrensPriceTest
    {
        private readonly IPrice CHILDRENS_PRICE = new ChildrensPrice();

        [Test]
        public void ShouldCalculateCorrectChargeForChildrensMovie()
        {
            Assert.AreEqual(1.5, CHILDRENS_PRICE.GetCharge(1));
            Assert.AreEqual(1.5, CHILDRENS_PRICE.GetCharge(2));
            Assert.AreEqual(1.5, CHILDRENS_PRICE.GetCharge(3));
            Assert.AreEqual(3.0, CHILDRENS_PRICE.GetCharge(4));
            Assert.AreEqual(4.5, CHILDRENS_PRICE.GetCharge(5));
        }
    }
}

File: Agile Development Practices/Code/Java/videoworld/src/unit/java/com/thoughtworks/videorental/domain/ChildrensPriceTest.java
package com.thoughtworks.videorental.domain;

import static junit.framework.Assert.*;

import org.junit.Test;

public class ChildrensPriceTest {
	private static final Price CHILDRENS_PRICE = new ChildrensPrice();

	@Test
	public void shouldCalculateCorrectChargeForChildrensMovie() throws Exception {
		assertEquals(1.5, CHILDRENS_PRICE.getCharge(1));
		assertEquals(1.5, CHILDRENS_PRICE.getCharge(2));
		assertEquals(1.5, CHILDRENS_PRICE.getCharge(3));
		assertEquals(3.0, CHILDRENS_PRICE.getCharge(4));
		assertEquals(4.5, CHILDRENS_PRICE.getCharge(5));
	}
}
